extra_data/reader.py

```diff
@@ -61,6 +61,10 @@
 INDEX_DATA = 'INDEX'
 METADATA = 'METADATA'
 
+def initializer():
+    # prevent child processes from receiving KeyboardInterrupt
```

Comment:
I think a bit more clarity is useful now that this is separated from where it's used.

```suggestion
def ignore_sigint():
    # Used in child processes to prevent them from receiving KeyboardInterrupt
```